src/gui/DatabaseOpenWidget.cpp

```diff
@@ -194,6 +183,26 @@ void DatabaseOpenWidget::openDatabase()
     bool ok = m_db->open(m_filename, masterKey, &error, false);
     QApplication::restoreOverrideCursor();
     if (!ok) {
+        if (m_ui->editPassword->text().isEmpty() && !m_retryUnlockWithEmptyPassword) {
+            QScopedPointer<QMessageBox> msgBox(new QMessageBox(this));
+            msgBox->setIcon(QMessageBox::Critical);
+            msgBox->setWindowTitle(tr("Unlock failed and no password given"));
+            msgBox->setText(tr("Unlocking the database failed and you did not enter a password.\n"
+                               "Do you want to retry with an \"empty\" password instead?\n\n"
+                               "If your database really has no password and this keeps happening,\n"
+                               "go to \"Database Settings / Security\" and reset your password there."));
```

Comment:
Rephrase this to:
> To prevent this error from appearing you must go to "Database Settings / Security" and reset your password.

Reply:
This error is specific for when there is no password provided. I kinda wanted to include that into the description to make it more explicit for the user.

Reply:
We could even remove the message and retry implicitly. We'd effectively hide the different between an unset and an empty password this way and the user would never notice. Only problem: unlocking takes twice as long and they YubiKey will be challenged twice.

Reply:
Sorry I meant just rephrase the last sentence. Keep everything else.